### PR TITLE
Bump Rubocop to 0.49.1

### LINKED ISSUE #6089: Update dependency constraint to allow for rubocop v0.49.0
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.48.1`, and the latest version available is `0.49.0`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### LINKED ISSUE #6101: Update dependency constraint to allow for rubocop v0.49.1
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.48.1`, and the latest version available is `0.49.1`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### PR DESCRIPTION
**Pull request by ayastreb**
Upgrade Rubocop to 0.49.0 

Fix #6089
Fix #6101

### CODE REVIEW

**Comment by DirtyF** [comment]
@ayastreb Rubocop 0.49.1 is already out, should we bump?

**Comment by ayastreb** [comment]
Bumped 👍

File: lib/jekyll/commands/serve.rb [hunk 1]
```diff
@@ -136,7 +136,7 @@ def server_address(server, options = {})
 
         private
         def format_url(ssl_enabled, address, port, baseurl = nil)
-          format("%{prefix}://%{address}:%{port}%{baseurl}", {
+          format("%<prefix>s://%<address>s:%<port>i%<baseurl>s", {
```
Comment (pathawks) [inline lib/jekyll/commands/serve.rb]: I’ve never seen this before. Type safety?

Comment (ayastreb) [inline lib/jekyll/commands/serve.rb]: It's from this cop - https://github.com/bbatsov/rubocop/blob/master/manual/cops_style.md#styleformatstringtoken
by default it enforces annotated tokens (e.g. `%<prefix>s`), but can be confgiured to enforce template tokens (e.g. `%{prefix}`).

Which one do you think we should use?

**Comment by ashmaroli** [comment]
Fixes #6089, #6101

**Comment by ashmaroli** [comment]
> apparently this does not work for dependency issues

Probably because those lines were not issued by the **PR-author**..? A bug nevertheless..

**Comment by ayastreb** [comment]
Cool, I will try 😎 
Fixes #6089, #6101

File: .rubocop.yml [hunk 2]
```diff
@@ -9,6 +9,39 @@ AllCops:
     - benchmark/**/*
     - script/**/*
     - vendor/**/*
+Layout/AlignArray:
+  Enabled: false
+Layout/AlignHash:
+  EnforcedHashRocketStyle: table
+Layout/AlignParameters:
+  Enabled: false
+  EnforcedStyle: with_fixed_indentation
```
Comment (parkr) [inline .rubocop.yml]: Is the enforced style useless because we have it disabled?

File: .rubocop.yml [hunk 3]
```diff
@@ -9,6 +9,39 @@ AllCops:
...
+Layout/EmptyLinesAroundAccessModifier:
+  Enabled: false
+Layout/EmptyLinesAroundModuleBody:
+  Enabled: false
```
Comment (parkr) [inline .rubocop.yml]: We should probably enable this.

File: .rubocop.yml [hunk 4]
```diff
@@ -9,6 +9,39 @@ AllCops:
...
+Layout/IndentHash:
+  EnforcedStyle: consistent
+Layout/IndentHeredoc:
+  Enabled: false
+Layout/MultilineMethodCallIndentation:
+  EnforcedStyle: indented
+Layout/MultilineOperationIndentation:
+  EnforcedStyle: indented
+Layout/SpaceAroundOperators:
+  Enabled: false
```
Comment (parkr) [inline .rubocop.yml]: Let's enable this.

**Comment by parkr** [comment]
@jekyllbot: merge +dev

**Pull request merged**